src/manager/evm.rs

```diff
@@ -0,0 +1,185 @@
+// Copyright 2022-2023 Protocol Labs
+// SPDX-License-Identifier: MIT
+use std::collections::HashMap;
+use std::str::FromStr;
+use std::sync::{Arc, RwLock};
+
+use anyhow::{anyhow, Result};
+use async_trait::async_trait;
+use cid::Cid;
+use ethers::prelude::k256::ecdsa::SigningKey;
+use ethers::prelude::k256::Secp256k1;
+use ethers::prelude::{Signer, SignerMiddleware};
+use ethers::providers::{Authorization, Http, Middleware, Provider};
+use ethers::signers::{LocalWallet, Wallet};
+use fil_actors_runtime::types::{InitExecParams, InitExecReturn, INIT_EXEC_METHOD_NUM};
+use fil_actors_runtime::{builtin::singletons::INIT_ACTOR_ADDR, cbor};
+use fvm_shared::clock::ChainEpoch;
+use fvm_shared::METHOD_SEND;
+use fvm_shared::{address::Address, econ::TokenAmount, MethodNum};
+use ipc_gateway::{BottomUpCheckpoint, PropagateParams, WhitelistPropagatorParams};
+use ipc_sdk::subnet_id::SubnetID;
+use ipc_subnet_actor::{types::MANIFEST_ID, ConstructParams, JoinParams};
+use url::Url;
+
+use crate::config::Subnet;
+use crate::jsonrpc::{JsonRpcClient, JsonRpcClientImpl};
+use crate::lotus::client::LotusJsonRPCClient;
+use crate::lotus::message::ipc::SubnetInfo;
+use crate::lotus::message::mpool::MpoolPushMessage;
+use crate::lotus::message::state::StateWaitMsgResponse;
+use crate::lotus::message::wallet::WalletKeyType;
+use crate::lotus::LotusClient;
+
+use super::subnet::SubnetManager;
+
+type MiddlewareImpl = SignerMiddleware<Provider<Http>, Wallet<SigningKey>>;
+
+pub struct EthSubnetManager<M: Send + Sync> {
```

Comment:
I think we can exclude Send Sync in the struct as we are confining it in the impl

Reply:
Good point, fixed!